exporters/ostream/include/opentelemetry/exporters/ostream/log_exporter.h

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "opentelemetry/nostd/type_traits.h"
+#include "opentelemetry/sdk/logs/exporter.h"
+#include "opentelemetry/sdk/logs/log_record.h"
+#include "opentelemetry/version.h"
+
+#include <iostream>
+#include <sstream>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter
+{
+namespace logs
+{
+/**
+ * The OStreamLogExporter exports logs through an ostream (default set to std::cout)
+ */
+class OStreamLogExporter final : public opentelemetry::sdk::logs::LogExporter
+{
+public:
+  /**
+   * Create an OStreamLogExporter. This constructor takes in a reference to an ostream that the
+   * Export() method will send log data into. The default ostream is set to stdout.
+   */
+  explicit OStreamLogExporter(std::ostream &sout = std::cout) noexcept;
+
+  /*********************************** Helper functions ************************/
+
+  /*
+    print_value is used to print out the value of an attribute within a vector.
+    These values are held in a variant which makes the process of printing them much more
+    complicated.
+  */
+
+  template <typename T>
+  void print_value(const T &item)
+  {
+    sout_ << item;
+  }
+
+  template <typename T>
+  void print_value(const std::vector<T> &vec)
+  {
+    sout_ << '[';
+    size_t i  = 1;
+    size_t sz = vec.size();
+    for (auto v : vec)
+    {
+      sout_ << v;
+      if (i != sz)
+        sout_ << ',' << ' ';
+      i++;
+    };
+    sout_ << ']';
+  }
+
+// Prior to C++14, generic lambda is not available so fallback to functor.
+#if __cplusplus < 201402L
+
+  class OwnedAttributeValueVisitor
+  {
+  public:
+    OwnedAttributeValueVisitor(OStreamLogExporter &exporter) : exporter_(exporter) {}
+
+    template <typename T>
+    void operator()(T &&arg)
+    {
+      exporter_.print_value(arg);
+    }
+
+  private:
+    OStreamLogExporter &exporter_;
+  };
+
+#endif
+
+  void print_value(sdk::common::OwnedAttributeValue &value)
+  {
+#if __cplusplus < 201402L
+    nostd::visit(OwnedAttributeValueVisitor(*this), value);
+#else
+    nostd::visit([this](auto &&arg) { print_value(arg); }, value);
+#endif
+  }
+
+  void printMap(std::unordered_map<std::string, sdk::common::OwnedAttributeValue> map)
+  {
+    size_t size = map.size();
+    size_t i    = 1;
```

Comment:
Do we need output `{}` to wrap the whole map, like `{{key: value}, {key1: value1}}`, or just as `{key: value, key1: value1}`?

Reply:
the outer `{}` around the map is done in the main `Export()` function [here](https://github.com/open-telemetry/opentelemetry-cpp/pull/430/files#diff-5bab08f8a94686dae0568791dbdee4c097ead2922ae0c855436326dc52ccf145R79)

Reply:
This is the [expected output](https://github.com/open-telemetry/opentelemetry-cpp/pull/430/files#diff-30ea5c00a60dd17b4b14adefe2b4bee8d41888ef8afc74703862e9b3514d19f6R220-R221) I imagined

Reply:
Ok, the caller outputs `{}` and assumed its content is map. Better to output `{}` in `printMap` in case it is called somewhere else.

Reply:
Good call, will make that change